public/EditorPage.tsx

```diff
@@ -0,0 +1,895 @@
+                            key={faceIndex}
+                            className={` rounded-lg cursor-pointer hover:bg-slate-100 hover:border-slate-500 border-2 border-inherit flex justify-center active:scale-90 transition-transform ease-in-out ${isThisItemTheSelectedOne ? "bg-slate-200 hover:border-slate-500 " : ""}`}
+                            onClick={() => {
+                              updateStores({
+                                faceConfig: faceConfigCopy,
+                                faceIndex,
+                                sectionIndex,
+                                stateStoreProps,
+                                overrideList,
+                              });
+                            }}
+                          >
+                            <Face
+                              ref={overrideToRun.ref}
+                              faceConfig={faceConfigCopy}
+                              width={
+                                gallerySize == "md"
+                                  ? 100
+                                  : gallerySize == "sm"
+                                    ? 50
+                                    : 150
+                              }
+                              lazyLoad={true}
+                            />
+                          </div>
+                        );
+                      },
+                    )}
+                  </div>
+                )}
+              </div>
+            );
+          },
+        )}
+    </div>
+  );
+};
+
+const copyStringToClipboard = async (str: string) => {
+  try {
+    await navigator.clipboard.writeText(str);
+  } catch (err) {
+    console.error("Failed to copyStringToClipboard: ", err);
+  }
+};
+
+const EditorPageTopBar = () => {
+  const stateStore = useStateStore();
+  let {
+    faceConfig,
+    gallerySize,
+    setGallerySize,
+    gallerySectionConfigList,
+    shuffleGenderSettingObject,
+    shuffleRaceSettingObject,
+    setShuffleGenderSettingObject,
+    setShuffleRaceSettingObject,
+  } = stateStore;
+  const navigate = useNavigate();
+
+  const genders = ["male", "female"];
+  const races = ["white", "black", "brown", "asian"];
+
+  const [genderInvalidOptions, setGenderInvalidOptions] =
+    useState<boolean>(false);
+  const [raceInvalidOptions, setRaceInvalidOptions] = useState<boolean>(false);
+
+  const optionsNotDisplayed =
+    genders.length +
+    races.length -
+    (shuffleGenderSettingObject.length + shuffleRaceSettingObject.length);
+
+  return (
+    <div className="bg-slate-800 text-white flex justify-between w-full fixed z-50	">
+      <div className="flex gap-4 text-xl p-2 justify-around items-center">
+        <span className="cursor-pointer rounded-full p-1 m-0.5 hover:bg-slate-50 hover:text-slate-900">
+          <House weight="fill" size={24} onClick={() => navigate("/")} />
+        </span>
+        <span className="hidden md:inline">faces.js Editor</span>
+        <Badge
```

Comment:
At a glance I'm not sure exactly where the code is for it... but that little circle with a number in it next to the settings... I don't know what that's supposed to represent, so maybe it should just be deleted.